app/services/secure_headers_whitelister.rb

```diff
@@ -0,0 +1,18 @@
+class SecureHeadersWhitelister
+  def run
+    whitelisted_domains = domains(acs_urls(SERVICE_PROVIDERS.values))
+    SecureHeaders::Configuration.override(:saml) do |config|
+      config.csp[:form_action] += whitelisted_domains
```

Comment:
`#concat` is preferred to `#+` because it is more efficient. `#+` creates a new array whereas `#concat` appends the elements of the other array to `self`. The difference in speed is not huge, but it's still there. Might as well use the fastest, most efficient method:

```
Comparison:
    concat:  2333649.9 i/s
      plus:  2188289.2 i/s - same-ish: difference falls within error
```